pkg/utils/node/predicates.go

```diff
@@ -31,7 +31,28 @@ func IsReadyAndSchedulable(node v1.Node) bool {
 	return false
 }
 
-func IsPastEmptyTTL(node *v1.Node) bool {
+func IsReady(node v1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			return condition.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func FailsToBecomeReady(node v1.Node) bool {
+	if time.Since(node.GetCreationTimestamp().Time) < (15 * time.Minute) {
+		return false
+	}
+	for _, condition := range node.Status.Conditions {
```

Comment:
maybe we should make a helper func that puts these condition in a map or something so we don't have to do this ugly for-loop thing through here.

Reply:
I agree. I added a helper to reduce some lines, lemme know what you think.

Reply:
If only https://github.com/knative/pkg/blob/main/apis/condition_set.go